docs/content/preview/api/ysql/the-sql-language/statements/dml_insert.md

```diff
@@ -92,9 +122,34 @@ DO NOTHING | DO UPDATE SET *update_item* [ , ... ] [ WHERE *condition* ]
 
 #### *condition*
 
+An expression that returns a value of type boolean. 
+All rows are locked when the `ON CONFLICT DO UPDATE` action is taken, but only rows where this expression returns true get updated.
+
+This condition is evaluated after a conflict has been identified as a candidate to update.
+
+Only `NOT DEFERRABLE` constraints and unique indexes are supported as arbiters.
+`exclusion` constraints are not supported as arbiters with `ON CONFLICT DO UPDATE`. 
+
+The `ON CONFLICT DO UPDATE` clause is a _deterministic_ statement. 
+The command is not allowed to update any existing row more than once, and doing so raises a `cardinality violation` error.
+Rows to be inserted must not conflict with each other on the used constraint or arbiter index.
+
+`ON CONFLICT DO UPDATE` does not support updating the partition key of a conflicting row in a partitioned table that results in the row moving to a new partition.
+
+{{< note title="Tip" >}}
+
+It is recommended to use a unique index for `ON CONFLICT` instead of a constraint, because you can replace the underlying unique index with another more or less equivalent index before dropping the index being replaced.
+
+{{< /note >}}
+
 ### Compatibility
 
-The SQL standard specifies that `OVERRIDING SYSTEM VALUE` can only be specified if an identity column that is generated always exists. YugabyteDB allows the clause in any case and ignores it if it is not applicable.
+There are several PostgreSQL extensions to the SQL standard inherited in YugabyteDB as follows:
```

Comment:
Nit: I am not sure if the general goal for this section is to cover SQL Standard compatibility or PG compatibility. But maybe this is a more general question for the docs team. For SQL standard perspective this looks good to me (though not 100% sure of all details).